src/scipp/core/binning.py

```diff
@@ -488,8 +488,21 @@ def _make_groups(x, arg):
     _require_coord(arg, coord)
     if coord.bins is not None:
         coord = coord.copy().bins.constituents['data']
+    if coord.dtype in (_cpp.DType.int32, _cpp.DType.int64):
+        min_ = coord.min().value
+        max_ = coord.max().value
+        values = coord.values
+        pivot = len(values) // 100
+        unique = np.unique(values[:pivot])
+        if unique.min() != min_ or unique.max() != max_:
```

Comment:
Actually this still misses a check ensuring all intermediate values are there, i.e., `unique` must be a range of stepsize 1. Will update later.

Reply:
Fixed, timings not affected.